src/main/java/at/hannibal2/skyhanni/features/event/anniversary/Year400Features.kt

```diff
@@ -0,0 +1,167 @@
+        val correctPlayers = playerColors.filter { it.value == colorInHand }.keys
+
+        for (entity in EntityUtils.getEntities<EntityOtherPlayerMP>()) {
+            if (entity.isNpc()) continue
+            val color = if (entity.entityId in correctPlayers) correctColor else LorenzColor.DARK_GRAY
+            entity.setColor(color, colorInHand)
+        }
+    }
+
+    @HandleEvent(onlyOnSkyblock = true)
+    fun onPlayerSpawn(event: MobEvent.Spawn.Player) {
+        val entity = event.mob.baseEntity
+        DelayedRun.runDelayed(1.seconds) {
+            addPlayer(entity)
+        }
+    }
+
+    private fun addPlayer(entity: EntityLivingBase) {
+        val displayName = entity.displayName.formattedText
+        val colorCode = playerColorNametagPattern.matchMatcher(displayName) {
+            group("color")
+        } ?: run {
+            if (colorInHand != null) {
+                entity.setColor(LorenzColor.DARK_GRAY, null)
+            }
+            return
+        }
+
+        val color = colorCode.toCharArray().first().toLorenzColor()
+        val cakeColor = CakeColor.entries.find { it.color == color } ?: run {
+            ErrorManager.logErrorStateWithData(
+                "Unknown slice of cake color",
+                "Unknown slice of cake color",
+                "displayName" to displayName,
+                "colorCode" to colorCode,
+                "color" to color,
+            )
+            return
+        }
+        playerColors[entity.entityId] = cakeColor
+
+        colorInHand?.let {
+            entity.setColor(cakeColor.color, it)
+        }
+    }
+
+    private fun EntityLivingBase.setColor(color: LorenzColor, currentHand: CakeColor?) {
+        RenderLivingEntityHelper.setEntityColor(this, color.toColor().addAlpha(1)) { config.teamFinder && colorInHand == currentHand }
+    }
+
+    @HandleEvent
+    fun onRealPlayerDeSpawnEvent(event: MobEvent.DeSpawn.Player) {
+        playerColors.remove(event.mob.baseEntity.entityId)
+    }
+
+    @HandleEvent
+    fun onPunch(event: EntityClickEvent) {
+        val entity = event.clickedEntity
+        if (colorInHand == null) return
+        if (entity !is EntityOtherPlayerMP) return
+        if (entity.isNpc()) return
+
+        lastClickedPlayer = entity.entityId
+        lastClickedPlayerTime = SimpleTimeMark.now()
+    }
+
+    @HandleEvent
+    fun onSystemMessage(event: SystemMessageEvent) {
+        if (!fatPlayerMessagePattern.matches(event.message.removeColor())) return
+        if (lastClickedPlayerTime.passedSince() >= 500.milliseconds) return
+
+        val lastPlayer = lastClickedPlayer ?: return
+        playerColors[lastPlayer] = null
+        lastClickedPlayer = null
+        lastClickedPlayerTime = SimpleTimeMark.farPast()
+
+        EntityUtils.getEntityByID(lastPlayer)?.let {
+            if (it is EntityLivingBase) {
+                it.setColor(LorenzColor.DARK_GRAY, colorInHand)
+            }
+        }
```

Comment:
If you would have saved the the mob it would be:
(Also the alpha should be set to 254 if you want to have it full, read comment on Mob.highlight)
```suggestion
        lastPlayer.highlight(LorenzColor.DARK_GRAY, colorInHand)
```